pullreq_pipeline.go

```diff
@@ -5,25 +5,86 @@ import (
 	"io/ioutil"
 	"os"
 	"strconv"
+	"strings"
 
 	"github.com/google/go-github/v28/github"
+	clientgitlab "github.com/mendersoftware/integration-test-runner/client/gitlab"
 	"github.com/mendersoftware/integration-test-runner/git"
 	"github.com/sirupsen/logrus"
+	"github.com/xanzy/go-gitlab"
 )
 
-func createPullRequestBranch(log *logrus.Entry, pr *github.PullRequestEvent, conf *config) error {
+func startPRPipeline(log *logrus.Entry, ref string, event *github.PullRequestEvent, conf *config) error {
+	client, err := clientgitlab.NewGitLabClient(
+		conf.gitlabToken,
+		conf.gitlabBaseURL,
+		conf.dryRunMode,
+	)
+	if err != nil {
+		return err
+	}
+	pr := event.GetPullRequest()
+	org := event.GetOrganization().GetLogin()
+	head := pr.GetHead()
+	base := pr.GetBase()
+	repo := event.GetRepo()
+	repoURL, err := getRemoteURLGitLab(org, repo.GetName())
+	if err != nil {
+		return err
+	}
+	repoHostURI := strings.SplitN(repoURL, ":", 2)
+	if len(repoHostURI) != 2 {
+		return fmt.Errorf("invalid GitLab URL '%s': failed to start gitlab pipeline", repoURL)
+	}
+	gitlabPath := repoHostURI[1]
+
+	pipeline, err := client.CreatePipeline(gitlabPath, &gitlab.CreatePipelineOptions{
+		Ref: &ref,
+		Variables: []*gitlab.PipelineVariable{{
+			Key:   "CI_EXTERNAL_PULL_REQUEST_IID",
+			Value: strconv.Itoa(event.GetNumber()),
+		}, {
+			Key:   "CI_EXTERNAL_PULL_REQUEST_SOURCE_REPOSITORY",
+			Value: head.GetRepo().GetFullName(),
+		}, {
+			Key:   "CI_EXTERNAL_PULL_REQUEST_TARGET_REPOSITORY",
+			Value: repo.GetFullName(),
+		}, {
+			Key:   "CI_EXTERNAL_PULL_REQUEST_SOURCE_BRANCH_NAME",
+			Value: head.GetRef(),
+		}, {
+			Key:   "CI_EXTERNAL_PULL_REQUEST_SOURCE_BRANCH_SHA",
+			Value: head.GetSHA(),
+		}, {
+			Key:   "CI_EXTERNAL_PULL_REQUEST_TARGET_BRANCH_NAME",
+			Value: base.GetRef(),
+		}, {
+			Key:   "CI_EXTERNAL_PULL_REQUEST_TARGET_BRANCH_SHA",
+			Value: base.GetSHA(),
+		}},
+	})
+	if err != nil {
+		return err
+	} else {
+		log.Debugf("started pipeline for PR: %s", pipeline.WebURL)
+	}
+
+	return nil
+}
+
+func syncPullRequestBranch(log *logrus.Entry, pr *github.PullRequestEvent, conf *config) (string, error) {
 
 	action := pr.GetAction()
 	if action != "opened" && action != "edited" && action != "reopened" &&
 		action != "synchronize" && action != "ready_for_review" {
```

Comment:
These are now checked by the caller, so we could remove from here. Don't harm neither, so do as you wish.